pkg/ebpf/c/capture_filtering.h

```diff
@@ -93,15 +99,41 @@ statfunc bool filter_file_type(void *ctx, void *filter_map, size_t map_idx, stru
             if (pipe != NULL) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_SOCKET_FILES) {
+        }
+        if (*type_filter & FILTER_SOCKET_FILES) {
             if (imode_mode & S_IFSOCK) {
                 type_filter_match = true;
             }
-        } else if (*type_filter & FILTER_NORMAL_FILES) {
+        }
+        if (*type_filter & FILTER_NORMAL_FILES) {
             if (imode_mode & S_IFREG) {
                 type_filter_match = true;
             }
         }
+        if (*type_filter & FILTER_ELF_FILES) {
+            file_id_t file_id = get_file_id(file);
+            file_id.ctime = 0;
+            if (start_pos == 0) {
+                // Check if header is matching ELF header
+                u8 header[FILE_MAGIC_HDR_SIZE] = {};
```

Comment:
Shouldn't this array be initialized to 0?
This can be important if header size is less than `FILE_MAGIC_HDR_SIZE`

Reply:
So using the syntax of `x = {}` should initialized the value to zero as far as I know.
I have checked it once with @roikol and it worked.